syntax = "proto3";

package masterworker;

service WorkerImpl {
    rpc Map (Shard) returns (MapReply) {}
    rpc Reduce (ReduceFile) returns (ReduceReply) {}
}

message Shard {
    repeated ShardFileInfo files = 1;
    int32 nOutputFiles = 2;
}

message ShardFileInfo {
    string fileName = 1;
    int32 start = 2;
    int32 finish = 3;
}

message MapReply {
    string mapFile = 1;
}

message ReduceFile {
    string fileName = 1;
}

message ReduceReply {
    string message = 1;
}